skbio/alignment/_tabular_msa.py

```diff
@@ -716,8 +730,119 @@ def reindex(self, key=None, keys=None):
             keys_.flags.writeable = False
 
         self._keys = keys_
+        self._minter = minter
+
+    def _fail_if_duplicate_keys(self, keys):
```

Comment:
Can you rename this to `_assert_valid_keys`? This will match the style we've used elsewhere (e.g., `Sequence._assert_can_cast_to`). The name is also more general because we're validating uniqueness *and* hashability (and in the future this method could place additional constraints on keys).